packages/react-i18n/src/errors.ts

```diff
@@ -1,12 +1,45 @@
-export class MissingTranslationError extends Error {}
-export class MissingReplacementError extends Error {}
-export class MissingCurrencyCodeError extends Error {}
-export class MissingCountryError extends Error {}
-export class InvalidI18nConnectionError extends Error {}
+import {Replacements} from './types';
+
+export class MissingTranslationError extends Error {
+  constructor(key: string) {
+    super(`Missing translation for key: ${key}`);
+  }
+}
+export class MissingReplacementError extends Error {
+  constructor(replacement: string, replacements: Replacements = {}) {
+    const errorMessages = [`No replacement found for key '${replacement}'.`];
+    const replacementKeys = Object.keys(replacements);
+
+    if (replacementKeys.length < 1) {
+      errorMessages.push('And no replacements were passed in.');
```

Comment:
I think putting this in parens instead of a separate sentence reads better

Reply:
sure